common/djangoapps/track/utils.py

```diff
@@ -28,3 +30,38 @@ def default(self, obj):  # pylint: disable=method-hidden
             return obj.isoformat()
 
         return super(DateTimeJSONEncoder, self).default(obj)
+
+
+class LatinJSONEncoder(DateTimeJSONEncoder):
+    """
+    JSON encoder for UnicodeDecodeError with latin1 characters
+    """
+
+    def encode(self, obj):
+        """
+        encode method gets an original object
+        and returns result string. obj argument will be the
+        object that is passed to json.dumps function
+        """
+        obj = self.iterate_dictionary(obj)
+
+        return super(LatinJSONEncoder, self).encode(obj)
+
+    def iterate_dictionary(self, obj):
+        """
+        Iterate over nested dict structure
```

Comment:
Please give a description of what it does when it iterates over the structure.  (E.g. that it tests that all strings are decodeable as utf8.   If some string is not, it then tries instead to decode as latin1, and if succeeds, tries to "fix" by substituting the decoded string.)